dashboard/app/controllers/policy_compliance_controller.rb

```diff
@@ -0,0 +1,86 @@
+class PolicyComplianceController < ApplicationController
+  before_action :authenticate_user!, except: [:child_account_consent]
+
+  # GET /policy_compliance/child_account_consent
+  # URL where parents or guardians give consent for their child to have an
+  # account on Code.org
+  def child_account_consent
+    @permission_granted = false
+    token = request.params[:token]
+    return render status: :bad_request if token.blank?
+    #Get parent permission request
+    permission_request = ParentalPermissionRequest.find_by(uuid: token)
+    return render status: :bad_request if permission_request.nil?
+    #Get User
+    user = permission_request.user
+    #Update User
+    if user.child_account_compliance_state != User::ChildAccountCompliance::PERMISSION_GRANTED
+      user.update_child_account_compliance(User::ChildAccountCompliance::PERMISSION_GRANTED)
+      user.save!
+      parent_email = permission_request.parent_email
+      ParentMailer.parent_permission_confirmation(parent_email).deliver_now
+    end
+    @permission_granted = true
+    @permission_granted_date = user.child_account_compliance_state_last_updated
+  end
+
+  # POST /policy_compliance/child_account_consent
+  # This URL creates a parental permission request. It is invoked via the
+  # lockout panel or potentially user settings by the child account.
+  def child_account_consent_request
+    # If we already comply, don't suddenly invalid it
+    if current_user.child_account_compliance_state == User::ChildAccountCompliance::PERMISSION_GRANTED
+      redirect_back fallback_location: '/lockout' and return
```

Comment:
Please, don't use hardcoded paths, it's a bad practice
```suggestion
      return redirect_back fallback_location: lockout_path
```